### PR TITLE
[SYCL][Driver] Force driver to set -unknown- OS for regular SYCL offloading

### PR DESCRIPTION
**Pull request by AGindinson**
The only case that remains uncovered is the usage of -fsycl-device-only
(or --sycl). More refactoring will be needed to handle this.

Also, the code responsible for setting the SYCL device triples was refactored
for better re-usability.

Signed-off-by: Artem Gindinson

### CODE REVIEW

**Comment by AGindinson** [comment]
@agozillon FYI

File: clang/lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -3568,18 +3559,11 @@ class OffloadingActionBuilder final {
...
-        else
-          TT.setArch(llvm::Triple::spir64);
-        TT.setVendor(llvm::Triple::UnknownVendor);
-        TT.setOS(llvm::Triple(llvm::sys::getProcessTriple()).getOS());
-        TT.setEnvironment(llvm::Triple::SYCLDevice);
-        if (C.getDriver().IsCLMode())
-          TT.setObjectFormat(llvm::Triple::COFF);
+        const char *SYCLTargetArch = SYCLfpga ? "spir64_fpga" : "spir64";
+        llvm::Triple TT = C.getDriver().MakeSYCLDeviceTriple(SYCLTargetArch);
         SYCLTripleList.push_back(TT);
```
Comment (mdtoguchi) [inline clang/lib/Driver/Driver.cpp]: TT isn't necessary - move the MakeTriple into the push.

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 2]
```diff
@@ -5495,11 +5495,7 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
       }
     } else {
       // Use the default.
-      llvm::Triple TT(Triple);
-      TT.setArch(llvm::Triple::spir64);
-      TT.setVendor(llvm::Triple::UnknownVendor);
-      TT.setOS(llvm::Triple(llvm::sys::getProcessTriple()).getOS());
-      TT.setEnvironment(llvm::Triple::SYCLDevice);
+      llvm::Triple TT(C.getDriver().MakeSYCLDeviceTriple());
       TargetInfo += TT.normalize();
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/Clang.cpp]: Same as other comment, doesn't look like TT is really needed and just append the MakeTriple

**Comment by AGindinson** [comment]
> Do we have an idea if this change will impact testing (for tests that use -fsycl-targets explicitly)?

As I've discovered soon thereafter, these do affect one of the SYCL LIT tests. The latest commit aligns that for SYCL.

**Comment by mdtoguchi** [comment]
> > Do we have an idea if this change will impact testing (for tests that use -fsycl-targets explicitly)?
> 
> As I've discovered soon thereafter, it does affect one of the SYCL LIT tests. The latest commit aligns that for SYCL.

I saw new changes come in as I sent out the review.  I guess testing beyond LIT will be captured later and tests will need to be addressed then.

**Pull request merged**